### DISCUSSION TITLE
Can't install on Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by jluixjurado** [bytespider/Meross#74] [Q&A] [answered]
Hi there!
I'm trying to exec this tool but, when running 
     ./meross-info --verbose --include-wifi
It complains that I need uuid4 (Error: Cannot find module 'uuid4'). When I try npm install uuid4, it says: (npm ERR! notsup Required: {"node":">=18.0.0"})
So I use npm install uuid4 --force
Afer that, running ./meross-info --verbose --include-wifi I receive:
     /home/xxxxxxx/Meross/node_modules/string-kit/lib/unicode.js:237
                return emojiWidthLookup.get( code ) ?? 2 ;
                                                     ^

     SyntaxError: Unexpected token '?'

Here I'm stucked. Could you help me?

Thanks in advance and Happy Christmas!

**Comment by bytespider** [comment]
Hey and Happy Christmas to you too.

As per the error message, you need Node 18. You can do this with a Node manager like FNM or NVM, or you can install using nodesource's repo https://github.com/nodesource/distributions?tab=readme-ov-file#installation-instructions

**Comment by vahidirp** [comment, marked as answer]
Check Node.js Version:
Make sure you have a version of Node.js installed that is 14.0.0 or higher. You can check your Node.js version by running:
```
node -v
```

If you have an older version, consider upgrading Node.js. You can use a version manager like nvm (Node Version Manager) to easily switch between Node.js versions.

Upgrade Node.js:
If you need to upgrade Node.js, you can use nvm to install a newer version. First, install nvm by following the instructions at: https://github.com/nvm-sh/nvm#installation

Once nvm is installed, you can install the latest LTS version of Node.js:


```
nvm install --lts
```
Then, use the following command to set the LTS version as the default:
```
nvm use --lts
```
Verify that you are now using a version of Node.js that is 14.0.0 or higher:
```
node -v
```

Install Dependencies:
After upgrading Node.js, navigate to your project directory and reinstall the dependencies:

```
npm install
```
This should install the necessary dependencies for your project, and the error related to the nullish coalescing operator should be resolved.

After following these steps, try running your tool again:
```
./meross-info --verbose --include-wifi
```
